mldsa/src/sign.c

```diff
@@ -716,6 +730,7 @@ int crypto_sign_verify_internal(const uint8_t *sig, size_t siglen,
                                 int externalmu)
 {
   unsigned int i;
+  int res = 0;
```

Comment:
We shouldn't default to success. Change to `int res;` here and set `res = 0` once everything went through?